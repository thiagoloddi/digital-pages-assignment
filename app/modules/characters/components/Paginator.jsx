import React from 'react';
import injectSheet from 'react-jss';

const style = {
  container: {
    marginTop: '10px',
    display: 'inline-block',
    border: '1px solid #DCDCDC',
    height: '35px',
    '& div + div': {
      borderLeft: '1px solid #DCDCDC'
    },
    '& > div': {
      color: '#00ACED',
      display: 'inline-block',
      height: '100%',
      minWidth: '35px',
      lineHeight: '35px',
      cursor: 'pointer',
      padding: '0 5px',
      '&:hover': {
        backgroundColor: '#f5f5f5'
      },
      '&.-selected': {
        color: 'white',
        backgroundColor: '#00ACED'
      }
    }
  }
  
};

const Paginator = ({ classes, current, last, onClick }) => {
  const location = current > 5 ? current <= last -5 ? 'center' : 'right' : 'left'; 
  const labels = {
    left: [1, 2, 3, 4, 5, 6, "...", last - 1, last],
    center: [1, 2, "...", current - 1, current, current + 1, "...", last - 1, last],
    right: [1, 2, "...", last - 5, last - 4, last - 3, last - 2, last - 1, last]
  }

  return (
    <div className={classes.container}>
      <div onClick={onClick}>{"<<"}</div>
      <div onClick={onClick}>{"<"}</div>
      {labels[location].map((label, i) => 
        <div key={i} className={label == current ? '-selected' : ''} onClick={onClick}>{label}</div>
      )}
      <div onClick={onClick}>{">"}</div>
      <div onClick={onClick}>{">>"}</div>
    </div>
  );
}

export default injectSheet(style)(Paginator);