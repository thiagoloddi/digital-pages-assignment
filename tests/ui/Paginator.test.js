import React from 'react';
import renderer from 'react-test-renderer';
import Paginator from '../../app/modules/characters/components/Paginator';

const last = 50;


it('renders correctly first third', () => {
  const tree = renderer
    .create(<Paginator current={1} last={last} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly second third', () => {
  const tree = renderer
    .create(<Paginator current={25} last={last} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly last third', () => {
  const tree = renderer
    .create(<Paginator current={50} last={last} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});