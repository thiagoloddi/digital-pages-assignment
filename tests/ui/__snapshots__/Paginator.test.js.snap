// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly first third 1`] = `
<div
  className="Paginator-container-0-1-1"
>
  <div>
    &lt;&lt;
  </div>
  <div>
    &lt;
  </div>
  <div
    className="-selected"
  >
    1
  </div>
  <div
    className=""
  >
    2
  </div>
  <div
    className=""
  >
    3
  </div>
  <div
    className=""
  >
    4
  </div>
  <div
    className=""
  >
    5
  </div>
  <div
    className=""
  >
    6
  </div>
  <div
    className=""
  >
    ...
  </div>
  <div
    className=""
  >
    49
  </div>
  <div
    className=""
  >
    50
  </div>
  <div>
    &gt;
  </div>
  <div>
    &gt;&gt;
  </div>
</div>
`;

exports[`renders correctly last third 1`] = `
<div
  className="Paginator-container-0-1-1"
>
  <div>
    &lt;&lt;
  </div>
  <div>
    &lt;
  </div>
  <div
    className=""
  >
    1
  </div>
  <div
    className=""
  >
    2
  </div>
  <div
    className=""
  >
    ...
  </div>
  <div
    className=""
  >
    45
  </div>
  <div
    className=""
  >
    46
  </div>
  <div
    className=""
  >
    47
  </div>
  <div
    className=""
  >
    48
  </div>
  <div
    className=""
  >
    49
  </div>
  <div
    className="-selected"
  >
    50
  </div>
  <div>
    &gt;
  </div>
  <div>
    &gt;&gt;
  </div>
</div>
`;

exports[`renders correctly second third 1`] = `
<div
  className="Paginator-container-0-1-1"
>
  <div>
    &lt;&lt;
  </div>
  <div>
    &lt;
  </div>
  <div
    className=""
  >
    1
  </div>
  <div
    className=""
  >
    2
  </div>
  <div
    className=""
  >
    ...
  </div>
  <div
    className=""
  >
    24
  </div>
  <div
    className="-selected"
  >
    25
  </div>
  <div
    className=""
  >
    26
  </div>
  <div
    className=""
  >
    ...
  </div>
  <div
    className=""
  >
    49
  </div>
  <div
    className=""
  >
    50
  </div>
  <div>
    &gt;
  </div>
  <div>
    &gt;&gt;
  </div>
</div>
`;
